common/stringop.c

```diff
@@ -64,12 +64,8 @@ char *lenient_strncat(char *dest, const char *src, size_t len) {
 
 // strcmp that also handles null pointers.
 int lenient_strcmp(char *a, char *b) {
-	if (a == b) {
-		return 0;
-	} else if (!a) {
-		return -1;
-	} else if (!b) {
-		return 1;
+	if (!a || !b || a == b) {
+		return a - b;
```

Comment:
Not a fan of this. This casts a `size_t` into an `int`. (I'm not even sure the cast happens _after_ the subtraction)

Reply:
I looked it up, and this is undefined behaviour, so I will revert it.